lib/core/src/sdk.rs

```diff
@@ -731,52 +775,102 @@ impl LiquidSdk {
+                );
+
+                let Some(amount_sat) = liquid_address_data.amount_sat else {
+                    return Err(PaymentError::AmountOutOfRange);
+                };
+
+                let payer_amount_sat = amount_sat + fees_sat;
+                ensure_sdk!(
+                    payer_amount_sat <= self.get_info().await?.balance_sat,
+                    PaymentError::InsufficientFunds
+                );
 
-        let amount_sat = get_invoice_amount!(&req.invoice);
-        let payer_amount_sat = amount_sat + req.fees_sat;
+                self.send_direct_payment(&liquid_address_data.address, amount_sat, *fees_sat)
+                    .await
+            }
+            SendDestination::Bolt11 { invoice } => {
+                self.check_send_payment_invoice(&invoice.bolt11, *fees_sat)
+                    .await
+            }
+        }
+    }
+
+    async fn check_send_payment_invoice(
+        &self,
+        invoice: &str,
+        fees_sat: u64,
+    ) -> Result<SendPaymentResponse, PaymentError> {
+        self.ensure_send_is_not_self_transfer(invoice)?;
+        self.validate_invoice(invoice)?;
+
+        let amount_sat = get_invoice_amount!(invoice);
+        let payer_amount_sat = amount_sat + fees_sat;
         ensure_sdk!(
             payer_amount_sat <= self.get_info().await?.balance_sat,
             PaymentError::InsufficientFunds
         );
 
-        match self.swapper.check_for_mrh(&req.invoice)? {
+        match self.swapper.check_for_mrh(invoice)? {
             // If we find a valid MRH, extract the BIP21 amount and address, then pay via onchain tx
-            Some((address, amount_btc)) => {
-                self.send_payment_via_mrh(req, &address, amount_btc).await
+            Some((address, _amount_sat)) => {
+                info!("Found MRH for L-BTC address {address} and amount_sat {amount_sat}");
+                self.send_direct_payment(&address, amount_sat, fees_sat)
+                    .await
             }
 
             // If no MRH found, perform usual swap
-            None => self.send_payment_via_swap(req).await,
+            None => self.send_payment_via_swap(invoice, fees_sat).await,
         }
     }
 
-    /// Performs a Send Payment by doing an onchain tx to the L-BTC address in the MRH.
-    async fn send_payment_via_mrh(
+    /// Performs a Send Payment by doing an onchain tx to a L-BTC address
+    async fn send_direct_payment(
         &self,
-        req: &PrepareSendResponse,
         lbtc_address: &str,
-        amount_btc: f64,
+        receiver_amount_sat: u64,
+        fees_sat: u64,
     ) -> Result<SendPaymentResponse, PaymentError> {
-        let amount_sat: u64 = (amount_btc * 100_000_000.0) as u64;
-        info!("Found MRH for L-BTC address {lbtc_address} and amount_sat {amount_sat}");
-
-        let receiver_amount_sat = get_invoice_amount!(req.invoice);
+        // TODO Ensure that `None` provides the lowest fees possible (0.01 sat/vbyte)
```

Comment:
Which `None` does this refer to? And is this comment still relevant to the PR, or can be removed?

Reply:
It's relevant, it ties in to #423. Once we'll be able to broadcast via our own node we'll need a global method that can tell us exactly which fees to use any broadcast since we won't be limited by Boltz API